### PR TITLE
Log heap size regularly

### PR DESCRIPTION
**Pull request by johnduffell**
applications blew up a few times with no change in load, basically due to crazy gc
cloudwatch metrics only log every minute
This PR makes it log heap usage every 5 seconds, and hopefully logs the job starting/finishing more accurately in some cases.
heap size vs time spent in GC for applications:
image
There's still a little more work to make all the jobs use futures and log when they finish.
@rich-nguyen

### CODE REVIEW

File: common/app/common/jobs.scala [hunk 1]
```diff
@@ -41,22 +41,23 @@ object Jobs extends Logging {
...
   scheduler.start()
 
+  // TODO delete this function and make easy to understand function(s)
   def schedule(name: String, cron: String)(block: => Unit): Unit = {
-    schedule(name, CronScheduleBuilder.cronSchedule(new CronExpression(cron)))(block)
+    schedule(name, CronScheduleBuilder.cronSchedule(new CronExpression(cron)))(Future(block))// TODO make a version to take a future
   }
 
-  def schedule(name: String, cron: String, timeZone: TimeZone)(block: => Unit): Unit = {
+  def scheduleWeekdayJob(name: String, hour: Int, minute: Int, timeZone: TimeZone)(block: => Future[Unit]): Unit = {
```
Comment (rich-nguyen) [inline common/app/common/jobs.scala]: is `hour, minute` a bit confusing for a cron wrapper? Why not `minute, hour`?

File: common/app/common/metrics.scala [hunk 2]
```diff
@@ -187,11 +185,22 @@ trait CloudWatchApplicationMetrics extends GlobalSettings {
     Jobs.schedule("ApplicationSystemMetricsJob", "36 * * * * ?"){
       report()
     }
+    Jobs.scheduleEveryNSeconds("LogMetricsJob", 5) {
+      val heapUsed = bytesAsMb(ManagementFactory.getMemoryMXBean.getHeapMemoryUsage.getUsed)
```
Comment (rich-nguyen) [inline common/app/common/metrics.scala]: Prod only?

**Comment by johnduffell** [comment]
good points, I've updated - how does it look now @rich-nguyen ?

**Comment by rich-nguyen** [comment]
:+1:

File: common/app/common/metrics.scala [hunk 3]
```diff
@@ -185,16 +185,20 @@ trait CloudWatchApplicationMetrics extends GlobalSettings with Logging {
     Jobs.schedule("ApplicationSystemMetricsJob", "36 * * * * ?"){
       report()
     }
-    Jobs.scheduleEveryNSeconds("LogMetricsJob", 5) {
-      val heapUsed = bytesAsMb(ManagementFactory.getMemoryMXBean.getHeapMemoryUsage.getUsed)
-      log.info(s"heap used: $heapUsed")
-      Future.successful(())
+    if (!Play.isDev) {
```
Comment (rich-nguyen) [inline common/app/common/metrics.scala]: `Configuration.environment.isProd`

Comment (johnduffell) [inline common/app/common/metrics.scala]: I actually did it that way on purpose! but true code is less interesting

**Pull request merged**